packages/opentelemetry-web/src/utils.ts

```diff
@@ -230,3 +230,76 @@ export function parseUrl(url: string): HTMLAnchorElement {
   a.href = url;
   return a;
 }
+
+/**
+ * Get element XPath
+ * @param target - target element
+ * @param optimised - when id attribute of element is present the xpath can be
+ * simplified to contain id
+ */
+export function getElementXPath(target: any, optimised?: boolean) {
+  if (target.nodeType === Node.DOCUMENT_NODE) {
+    return '/';
+  }
+  let xpath = '';
+  const targetValue = getNodeValue(target, optimised);
+  if (optimised && targetValue.indexOf('@id') > 0) {
+    return targetValue;
+  }
+  if (target.parentNode) {
+    xpath += getElementXPath(target.parentNode, false);
+  }
+  xpath += targetValue;
+
+  return xpath;
```

Comment:
`?` is nothing more then `if` so you will have the same cyclometric complexity but sacrificing readability I also think that any of such `optimisation` is basically the task for obfuscators and uglifiers but here I want to be able to read that easily and know what is happening. For me `?` is for really simple stuff - just a personal preference :)

Reply:
What about something like:

```typescript
if (target.parentNode) {
  return getElementXPath(target.parentNode, false);
}
return targetValue;
```

This keeps the readability of simple if statements without having the xpath variable which is mutated unnecessarily.

Reply:
> What about something like:
> 
> ```ts
> if (target.parentNode) {
>   return getElementXPath(target.parentNode, false);
> }
> return targetValue;
> ```
> 
> This keeps the readability of simple if statements without having the xpath variable which is mutated unnecessarily.

It doesn't work at all

Reply:
Oh wow I don't know what I was thinking here I completely brain farted. This should work though and still avoids the `let`

```typescript
if (target.parentNode) {
  return `${getElementXPath(target.parentNode, false)}${targetValue}`;
}
return targetValue;
```

At the very least you should move the `let` down below the optimized if statement so that it is clear what the whole lifecycle of the var is from declaration to return in as small a space as possible.